src/code/templates/egg-game.js

```diff
@@ -101,44 +185,375 @@ var animationEvents = {
+        addChromAnimStage(BioLogica.MALE, defaultAnimationSpeed, '2', ['a', 'b']);
+        addChromAnimStage(BioLogica.MALE, defaultAnimationSpeed, 'XY', ['x', 'y']);
+        addGameteAnimStage(BioLogica.MALE, defaultAnimationSpeed);
+
+        if (debugTotalGameteCount !== 1) {
+          // randomize mother's chromosomes simultaneously
+          addGenomeAnimStage(BioLogica.FEMALE, enhancedAnimationSpeed);
+          addGameteAnimStage(BioLogica.FEMALE, enhancedAnimationSpeed);
+
+          // randomize father's chromosomes simultaneously
+          addGenomeAnimStage(BioLogica.MALE, enhancedAnimationSpeed);
+          addGameteAnimStage(BioLogica.MALE, enhancedAnimationSpeed);
+        }
+
+        // randomize the remaining gametes for both parents simultaneously
+        const totalGameteCount = debugTotalGameteCount || 8,
+              fullCycleCount = totalGameteCount - 2;
+        for (let i = 0; i < fullCycleCount; ++i) {
+          addGenomeAnimStage(BOTH_SEXES, enhancedAnimationSpeed);
+          addGameteAnimStage(BOTH_SEXES, enhancedAnimationSpeed);
+        }
+
+        stages.push({ type: 'complete' });
+      }
+
+      function selectStageChromosomes(speed, chroms) {
+        if (!_this) return;
+        let animatingGametes = cloneDeep(_this.state.animatingGametes) || [{}, {}, {}, {}];
+        chroms.forEach(({ sex, name }) => {
+          let gamete = animatingGametes && animatingGametes[sex],
+              side = gamete && gamete[name];
+          if (side) {
+            const parentId = sex === BioLogica.FEMALE ? 'mother' : 'father',
+                  chromContainerId = `${parentId}${name}${side}`,
+                  elt = document.getElementById(chromContainerId).parentNode;
+            // this will trigger an animation which will call onFinish() when complete,
+            // which will trigger the next stage of the animation.
+            _this.selectChromosomes(sex, speed, [{name, side, elt}],
+                                    animationEvents.randomizeChromosomes.onFinish);
+          }
+        });
+        animatingGametes[2] = clone(animatingGametes[0]);
+        animatingGametes[3] = clone(animatingGametes[1]);
+        // delay appearance of labels until the chromosomes arrive
+        setTimeout(() => _this.setState({ animatingGametes }), delayShowSelectedChromosomeLabels);
+      }
+
+      function toggleAnimatingGameteChromosome(sex, chromName, sides) {
+        let animatingGametes = _this.state.animatingGametes || [{}, {}, {}, {}],
+            gamete = animatingGametes && animatingGametes[sex],
+            currSide = gamete && gamete[chromName],
+            newSide = currSide != null
+                        ? (currSide === sides[0] ? sides[1] : sides[0])
+                        : oneOf(sides);
+        animatingGametes[sex] = assign({}, gamete, { [chromName]: newSide });
+        if (_this) _this.setState({ animatingGametes });
```

Comment:
I think that the `if (_this)` pattern is pretty close to the "isMounted" antipatern described here: https://facebook.github.io/react/blog/2015/12/16/ismounted-antipattern.html, where they recommend explicitly unsubscribing from any timers or listeners in `componentWillUnmount` instead.

Testing aborted animations (by changing the challenge hash in the url bar in the middle of animations), there are still a number of errors that get thrown, even with this check. For example, in this function, the `_this.state` above is null. Also, there are many cases where `setTimeout(() => [...] _this.setState` is used without checking `_this`.

I was worried that unsubscribing from all the animation events would be a really ugly task, and that this pattern would be better on balance, but actually your `timerSet` makes this really easy. As far as I could tell, I could remove all the instances of `if (_this)` and remove all the unmounting errors with just a few changes (which I could also put in a commit, here or in another branch, if that's easier):

1. Move `timerSet` to the top-level scope, and call `if (timerSet) timerSet.reset();` in `resetAnimationEvents`
2. Add a quick utility (at the top scope, or in /utilities) for remembering and clearing all timeouts:
```
var timers = [];
function _setTimeout(fn, delay) {
  timers.push(setTimeout(fn, delay));
}
function clearTimeouts() {
  for (let timer of timers) {
    clearTimeout(timer);
  }
}
```
3. Replace all `setTimeout` with `_setTimeout`, and call `clearTimeouts();` in `resetAnimationEvents`.

It's quite possible I missed something, but in testing I found that with those changes I could exit in the middle of any animation and not get errors.

Thoughts?